pint/pintk/plk.py

```diff
@@ -807,29 +954,74 @@ def canvasKeyEvent(self, event):
                 self.updatePlot(keepAxes=False)
                 self.call_updates()
         elif ukey == ord('d'):
+            #if any of the points are jumped, tell the user to delete the jump(s) first
+            jumped_copy = copy.deepcopy(self.jumped)
+            for param in self.psr.prefit_model.params:
+                if param.startswith("JUMP") and getattr(self.psr.prefit_model, param).frozen == True:
+                    self.updateJumped(param)
+            all_jumped = copy.deepcopy(self.jumped)
+            self.jumped = jumped_copy
+            if True in [a and b for a,b in zip(self.selected, all_jumped)]:
+                print("cannot delete jumped toas. Delete interfering jumps before deleting toas.")
+                return None
             #Delete the selected points
-            self.psr.toas.table = self.psr.toas.table[~self.selected].group_by('obs')
-            if hasattr(self.psr.toas, 'table_selects'):
-                for i in range(len(self.psr.toas.table_selects)):
-                    self.psr.toas.table_selects[i] = \
-                        self.psr.toas.table_selects[i][~self.selected].group_by('obs')
-            self.selected = np.zeros(self.psr.toas.ntoas, dtype=bool)
+            self.psr.fulltoas.table = self.psr.fulltoas.table[~self.selected].group_by('obs')
+            self.psr.toas = copy.deepcopy(self.psr.fulltoas)
+            if hasattr(self.psr.fulltoas, 'table_selects'):
+                for i in range(len(self.psr.fulltoas.table_selects)):
+                    self.psr.fulltoas.table_selects[i] = \
+                        self.psr.fulltoas.table_selects[i][~self.selected].group_by('obs')
+            self.jumped = self.jumped[~self.selected]
+            self.selected = np.zeros(self.psr.fulltoas.ntoas, dtype=bool)
             self.psr.update_resids()
             self.updatePlot(keepAxes=True)
             self.call_updates()
-        elif ukey == ord('s'):
-            #Apply the selection to TOAs object
-            self.psr.toas.select(self.selected)
-            self.selected = np.zeros(self.psr.toas.ntoas, dtype=bool)
-            self.psr.update_resids()
-            self.updatePlot(keepAxes=False)
-            self.call_updates()
         elif ukey == ord('u'):
             self.unselect()
+        elif ukey == ord('j'):
+            #jump the selected points, or unjump if already jumped
+            jump_name = self.psr.add_jump(self.selected)
```

Comment:
TODO added, although I think it might be better to just rename the function since the idea was one button to toggle a jump on and off. The function could also be split and then one button could call either resulting function, depending on the state of the system